store/store.go

```diff
@@ -227,3 +231,49 @@ func (s *store) ListServers(ctx context.Context, serviceID string) ([]*types.Rea
 	}
 	return servers, nil
 }
+
+func (s *store) Subscribe(subscriber func(), closeCh <-chan struct{}) {
+	options := &client.WatcherOptions{
+		Recursive: true,
+	}
+	watcher := s.kapi.Watcher(s.prefix, options)
+	go func() {
+		ctx, cancelFunc := context.WithCancel(context.Background())
+		defer cancelFunc()
+		respCh := make(chan *client.Response)
+		watcherFunc := func() error {
+			resp, err := watcher.Next(ctx)
+			if err == nil {
+				respCh <- resp
+			} else {
+				// stopCh closed
```

Comment:
What does this mean?

Reply:
I should rename `closeCh` to `stopCh`. If ctx.Err() != nil, it means something closed the stop channel, and the context got cancelled.

Reply:
I'll refactor this and break it up to try and make it clearer.